src/lib/TopLevel.hs

```diff
@@ -909,7 +936,57 @@ getFilteredLogger = do
   logger  <- getLogger
   return $ FilteredLogger shouldLog logger
 
--- -- === instances ===
+-- === hoisting Ix dicts ===
+
+hoistIxDicts
```

Comment:
Perhaps move this to Lower.hs?

Reply:
Yes, sure. I might even try moving it to simplification as we'd initially discussed and as you suggest below. The reason I didn't put it in simplification was that I realized that simplification calls `substM` on types rather than traversing them, so we wouldn't find all the ix dicts. But maybe I should just fix that.

Reply:
Right, I just want to avoid having TopLevel grow beyond the minimum. It's already a huge file and this really does not need to be there.

Reply:
Sure. I'm planning to move it to simplification in a follow-up patch. So let's leave it here until then.